pkg/cloud/vsphere/services/govmomi/vcenter/clone.go

```diff
@@ -122,7 +121,10 @@ func Clone(ctx *context.MachineContext, userData []byte) error {
 	if err != nil {
 		return errors.Wrapf(err, "error trigging clone op for machine %q", ctx)
 	}
-
+	_, err = task.WaitForResult(ctx, nil)
```

Comment:
Hi @pickledrick,

CAPV is reentrant and never blocks. If something isn't ready, it requeues the operation.